tests/functional/spec/navigation.spec.js

```diff
@@ -72,4 +75,73 @@ describe('Navigation', function() {
     expect(isViewSectionsVisible()).to.be.true
   })
 
+
+  it('Given a survey which has a "Summary" page, When I have not completed all the sections, Then I will not see a link called "Summary" in the navigation', function() {
+    // Given
+    openQuestionnaire('test_navigation.json')
+
+    // Then
+    let navigationLinks = Navigation.getNavigationLinkText()
+    expect(navigationLinks).to.not.contain('Summary')
+    expect(navigationLinks).to.not.contain('Submit answers')
+  })
+
+  it('Given a survey which has a "Summary" page, When I have completed all sections, Then I will see a link called "Summary" in the navigation', function() {
+    // Given
+    openQuestionnaire('test_navigation.json')
+
+    // When
+    Navigation.completeAllSections()
+
+    // Then
+    let navigationLinks = Navigation.getNavigationLinkText()
+    expect(navigationLinks).to.contain('Summary')
+    expect(navigationLinks).to.not.contain('Submit answers')
+  })
+
+   it('Given a survey which has a "Summary" page, When I click the link, Then I am taken to the summary screen', function() {
+    // Given
+    openQuestionnaire('test_navigation.json')
+
+    // When
+    Navigation.completeAllSections().clickSummaryLink()
+
+    // Then
+    expect(NavigationSummaryPage.isOpen()).is.true
+  })
+
+  it('Given a survey which has a "Confirmation" page, When I have not completed all the sections, Then I will not see a link called "Submit answers" in the navigation', function() {
```

Comment:
Yeah, sorry I meant unit tests :) Strictly speaking though, unit tests shouldn't be interacting with a schema file.